client/src/components/pages/RobotOverview/RobotOverview.jsx

```diff
@@ -60,6 +61,24 @@ const RobotOverview = () => {
     setSearchValue(value);
   };
 
+  /**
+   * @description Creates a new bot for the current userId
+   */
+  const createNewRobot = () => {
+    const robotName = 'New Robot';
+    createNewBot(userId, robotName)
+      .then((response) => response.json())
+      .then((newRobot) => {
+        const newRobotList = [...robotList]
+        newRobotList.unshift(newRobot);
+        setRobotList([]);
+        setRobotList(newRobotList);
```

Comment:
We don't need setRobotList([]) because setRobotList(newRobotList) just _replaces_ the current value of the state variable

Reply:
@lukashueller

Reply:
unfortunately, this is not the right case. When I remove this workaround, React only re-renders the last box in the GUI (adds a box at the back). Therefore, without this line of code, I don't get the desired result that the new robots are displayed first on the page. (As @SanJSp had requested)

Do you maybe have another idea how to solve the problem? Otherwise I would leave the workaround in there for now.
@WolfgangDaniel

Reply:
Interesting. I just looked it up and it seems like unshift might cause the problem. I would try writing the 4 lines as one line like this: setRobotList( robotList => [newRobot, ...robotList])
That should work. If you are intersted, here is a link to a medium article about that topic [](https://javascript.plainenglish.io/how-to-add-to-an-array-in-react-state-3d08ddb2e1dc)

Reply:
I was not able to fix it. Maybe lets try it in a bigger refactoring.